Lib/test/test_dataclasses.py

```diff
@@ -1451,6 +1452,29 @@ class A(types.GenericAlias):
         self.assertTrue(is_dataclass(type(a)))
         self.assertTrue(is_dataclass(a))
 
+    def test_is_dataclass_generic_class(self, expected=True):
+        T = TypeVar("T")
+        @dataclass
+        class B(Generic[T]):
+            a: T
+        @dataclass
+        class C(Generic[T]):
+            a: T
+            __class_getitem__ = classmethod(types.GenericAlias)
+
+        for cls in (B, C):
+            with self.subTest(cls=cls):
+                self.assertTrue(is_dataclass(cls))
+                self.assertEqual(is_dataclass(cls[int]), expected)
```

Comment:
Should not it always be true?

Add a test for class which subclasses `list` and inherits `__class_getitem__` from it.